### ISSUE TITLE
MDANSE does not launch on macOS 11.6 M1 chip

### ISSUE THREAD

**Issue by perenon** [ISISNeutronMuon/MDANSE#157]
Hello,
With Eric, we tested MDANSE on a macOS M1 machine, with system BigSur (11.6).

MDANSE crashed on startup with no error message.
However, using the provided python and lauching manually MDANSE gives:
```
$ /Applications/MDANSE.app/Contents/MacOS/python2
Python 2.7.18 (default, Oct 29 2020, 07:55:29) 
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import MDANSE
>>> from MDANSE.GUI.Apps import MainApplication
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Applications/MDANSE.app/Contents/Resources/lib/python2.7/MDANSE/GUI/Apps.py", line 19, in <module>
    from MDANSE.GUI.Plugins.PlotterPlugin import PlotterFrame
  File "/Applications/MDANSE.app/Contents/Resources/lib/python2.7/MDANSE/GUI/Plugins/PlotterPlugin.py", line 21, in <module>
    import netCDF4
  File "/Applications/MDANSE.app/Contents/Resources/lib/python2.7/site-packages/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
ImportError: dlopen(/Applications/MDANSE.app/Contents/Resources/lib/python2.7/site-packages/netCDF4/_netCDF4.so, 2): Library not loaded: @rpath/libnetcdf.13.dylib
  Referenced from: /Applications/MDANSE.app/Contents/Resources/lib/python2.7/site-packages/netCDF4/_netCDF4.so
  Reason: image not found
```

So we relocated netcdf lib
`$ install_name_tool -change @rpath/libnetcdf.13.dylib @executable_path/../Frameworks/libnetcdf.13.dylib /Applications/MDANSE.app/Contents/Resources/lib/python2.7/site-packages/netCDF4/_netCDF4.so`

`$ install_name_tool -change @rpath/libhdf5.103.dylib @executable_path/../Frameworks/libhdf5.103.dylib /Applications/MDANSE.app/Contents/Resources/lib/python2.7/site-packages/netCDF4/_netCDF4.so`

And we are facing now an issue with matplotlib
```
$ /Applications/MDANSE.app/Contents/MacOS/python2
Python 2.7.18 (default, Oct 29 2020, 07:55:29) 
[GCC 4.2.1 Compatible Apple LLVM 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from MDANSE.GUI.Apps import MainApplication
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Applications/MDANSE.app/Contents/Resources/lib/python2.7/MDANSE/GUI/Apps.py", line 19, in <module>
    from MDANSE.GUI.Plugins.PlotterPlugin import PlotterFrame
  File "/Applications/MDANSE.app/Contents/Resources/lib/python2.7/MDANSE/GUI/Plugins/PlotterPlugin.py", line 23, in <module>
    from matplotlib.figure import Figure
  File "/Applications/MDANSE.app/Contents/Resources/lib/python2.7/matplotlib/__init__.py", line 129, in <module>
    from . import cbook
ImportError: cannot import name cbook
```

We could eventually launch MDANSE after copying manually an older version of matplotlib, we can try to find the recipe if needed, but it was quite dirty.

If MDANSE could be launched, it crashed with a `line 6: 64243 Illegal instruction: 4` when using the VTK plugin.

So, to sum up all this as questions

**Are the relocations of libraries still active in the packaging process ?
Did you change the version of matplotlib, which could explain the second error ?
How did you install vtk on the build server ?**

Cheers
Remi

**Comment by RastislavTuranyi** [comment]
Hi Remi, not sure which MDANSE build you used, but it's most likely that it is one where the linking process has not been updated yet. The linking issues with the official release 1.5.1 have been fixed in #146, and the issues introduced on that branch have been corrected on branch [`147_fix_macos`](https://github.com/ISISNeutronMuon/MDANSE/tree/147_fix_macos).

The version of matplotlib has not changed, it has always been the latest version for Python 2.7; matplotlib 2.2.5. I am not sure how that issue came about; we have not encountered it before.

About VTK:

1. Originally, VTK 6.3.0 was installed with conda and then copied into the python used for MDANSE.
2. [This build](https://github.com/ISISNeutronMuon/MDANSE/actions/runs/2963575148) has VTK 6.3.0 built from source. However, the re-linking of dylibs is not correct, so you might want to use the `MDANSE/BuildServer/Unix/MacOS/relink-vtk.py` script on the `147_fix_macos` branch to do the linking.
3. [This build ](https://github.com/ISISNeutronMuon/MDANSE/actions/runs/2992773228) has VTK 9.1.0 built from source. The linking there is correct.

**Comment by RastislavTuranyi** [comment]
Yes, that's the version in which we discovered the issue with MacOS, #147. If I remember correctly, the opening and linking issues should have been resolved in [this build](https://github.com/ISISNeutronMuon/MDANSE/actions/runs/2926067067), after which we have been trying to resolve the VTK issue in further commits.

**Comment by perenon** [comment]
OK I'll try. It might be interesting to use our macOS 11 also.

**Comment by RastislavTuranyi** [comment]
You can't download the artifact anymore, only the installer (dmg file). Why would you need the artifacts? If you really need those, you would have to re-run that CI by clicking on the `re-run all jobs` button in the top right corner.

**Comment by perenon** [comment]
I confirm the application opens correctly.
However, it crashes as soon as we open the molecular viewer. Is it a bug you identified ?

**Comment by RastislavTuranyi** [comment]
That's exactly the issue I asked about in the meeting. The same thing happens when we try to plot with elevation, iso-surface, or scalar-field plotters, i.e. every part of MDANSE that uses VTK. I tried solving it on the [147_fix_macos](https://github.com/ISISNeutronMuon/MDANSE/tree/147_fix_macos) branch by changing the build conditions but so far unsuccessfully.

**Comment by perenon** [comment]
OK thanks. I did not catch this during the meeting.

- Is it specific to macOS ? To M1 chip ?
- Do you want me to change the name of the issue or to reopen a new one ?

**Comment by RastislavTuranyi** [comment]
1. It happens only on MacOS, but beyond that we are not sure; here are the facts: 
    1. Everything works fine on MacOS 10.x (not sure which version exactly) with intel processor
    2. MDANSE crashes at VTK parts on MacOS 12.5 with M1 processor
    3. MDANSE does not crash but the VTK parts are blank on MacOS 12.0 (beta) that is emulated on Windows 10 laptop with intel processor
    4. As you reported, MDANSE crashes at VTK parts on MacOS 11.6 with M1 processor
2. I think it's fine.

**Comment by RastislavTuranyi** [comment]
Actually, correction: I have just tested on Ubuntu 22 using the Ubuntu 22 installer and discovered that it crashes too. See #158 for more details.

**Comment by RastislavTuranyi** [comment]
I have just tested on my Windows 11 laptop and found that the Animation there crashes MDANSE (#159). It could be related but it's hard to say since the other parts that use VTK other than Animation work there.

**Comment by MBartkowiakSTFC** [comment]
This issue was caused by an incompatibility between wxWidgets and VTK. We addressed this issue by switching to PyQt in the Python 3 version.
